react/ChallengeTradePolicyCondition.tsx

```diff
@@ -36,18 +59,16 @@ const useProfileAllowed = (skip: boolean) => {
 }
 
 const ChallengeTradePolicyCondition: FC<Props> = ({
-  session,
   redirectPath = '/login',
   defaultContentVisibility = 'visible',
   children,
 }) => {
-  const unauthorized =
-    session && (session as SessionUnauthorized).type === 'Unauthorized'
-  const profileAllowed = useProfileAllowed(unauthorized)
+  const isAuthorized = useSessionAuthorization()
+  const profileAllowed = useProfileAllowed(!isAuthorized)
 
-  const isAuthenticated = unauthorized === true ? false : profileAllowed
+  const isAuthenticated = isAuthorized === false ? false : profileAllowed
```

Comment:
Is this the same as `const isAuthenticated = isAuthorized || profileAllowed`?

Reply:
`isAuthorized` could be `null`, so no.